cpg_workflows/filetypes.py

```diff
@@ -41,22 +42,18 @@ def __init__(
         if index_path:
             self.index_path = to_path(index_path)
             assert self.index_path.suffix == f'.{self.index_ext}'
-            self.full_index_suffix = str(self.index_path).replace(
-                str(self.path.with_suffix('')), ''
-            )
+            self.full_index_suffix = str(self.index_path).replace(str(self.path.with_suffix('')), '')
         self.reference_assembly = None
         if reference_assembly:
             self.reference_assembly = to_path(reference_assembly)
 
     @property
     @abstractmethod
-    def ext(self) -> str:
-        ...
+    def ext(self) -> str: ...
```

Comment:
I kind of wonder what it thinks it's doing here…

Could we avoid this by adding a docstring as is done for AlignmentInput.exists() above? e.g. (and ditto for `index_ext`)

```suggestion
    def ext(self) -> str:
        """The canonical extension for the file type, without a '.' at the start."""
```